core/ordersync_subprotocols.go

```diff
@@ -0,0 +1,129 @@
+package core
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/0xProject/0x-mesh/core/ordersync"
+	"github.com/0xProject/0x-mesh/zeroex"
+)
+
+// Ensure that FilteredPaginationSubProtocol implements the Subprotocol interface.
+var _ ordersync.Subprotocol = (*FilteredPaginationSubProtocol)(nil)
+
+// FilteredPaginationSubProtocol is an ordersync subprotocol which returns all orders by
+// paginating through them. It involves sending multiple requests until pagination is
+// finished and all orders have been returned.
+type FilteredPaginationSubProtocol struct {
+	app     *App
+	perPage int
+}
+
+// NewFilteredPaginationSubprotocol creates and returns a new FilteredPaginationSubprotocol
+// which will respond with perPage orders for each individual request/response.
+func NewFilteredPaginationSubprotocol(app *App, perPage int) *FilteredPaginationSubProtocol {
+	return &FilteredPaginationSubProtocol{
+		app:     app,
+		perPage: perPage,
+	}
+}
+
+// FilteredPaginationRequestMetadata is the request metadata for the
+// FilteredPaginationSubProtocol. It keeps track of the current page and SnapshotID,
+// which is expected to be an empty string on the first request.
+type FilteredPaginationRequestMetadata struct {
+	Page       int    `json:"page"`
+	SnapshotID string `json:"snapshotID"`
+}
+
+// FilteredPaginationResponseMetadata is the response metadata for the
+// FilteredPaginationSubProtocol. It keeps track of the current page and SnapshotID.
+type FilteredPaginationResponseMetadata struct {
+	Page       int    `json:"page"`
+	SnapshotID string `json:"snapshotID"`
+}
+
+// Name reutrns the name of the FilteredPaginationSubProtocol
+func (p *FilteredPaginationSubProtocol) Name() string {
+	return "/pagination-with-filter/version/0"
+}
+
+// GetOrders returns the orders for one page, based on the page number
+// and snapshotID in the corresponding to the given request. This is
+// the implementation for the "provider" side of the subprotocol.
+func (p *FilteredPaginationSubProtocol) GetOrders(ctx context.Context, req *ordersync.Request) (*ordersync.Response, error) {
```

Comment:
Would it make more sense to call this something like `getAndSendOrders`? I think it could be a little confusing that `HandleStream` is on the `provider` side and `GetOrders` is on the `requester` side in the `ordersync` protocol, but in this subprotocol, these names switch places.

Reply:
Changed the method names in the `ordersync.Subprotocol` interface. What do you think of the current version?

Reply:
It looks good!